File: klinara-ios/klinara-ios/Services/Notifications/MockNotificationsSeed.swift
import Foundation

/// Mock bildirim verisinin başlangıç durumu.
///
/// Tohum kasıtlı olarak **kurulumu yarım kalmış bir klinik**i temsil eder:
/// WhatsApp bağlı ama bir şablon Meta'da hâlâ onay bekliyor, bir hatırlatma
/// gönderilmiş, bir tanesi iletişim izni yüzünden `skipped` yazılmış, gelen
/// kutusunda okunmamış iki mesaj var. Her şeyin yolunda olduğu bir tohum,
/// ekranların asıl zor durumlarını (başarısız gönderim, atlanmış mesaj,
/// tanınmayan numara) hiç göstermezdi.
enum MockNotificationsSeed {

    static let messageReminderSent = "e1000000-0000-4000-8000-000000000001"
    static let messageConfirmationDelivered = "e1000000-0000-4000-8000-000000000002"
    static let messageBirthdaySkipped = "e1000000-0000-4000-8000-000000000003"
    static let messageReminderFailed = "e1000000-0000-4000-8000-000000000004"
    static let messageStaffInternal = "e1000000-0000-4000-8000-000000000005"

    static let inboxAyse = "e2000000-0000-4000-8000-000000000001"
    static let inboxUnknown = "e2000000-0000-4000-8000-000000000002"
    static let inboxHandled = "e2000000-0000-4000-8000-000000000003"

    static let optOutMehmet = "e3000000-0000-4000-8000-000000000001"

    /// Kiracının kendi metnini yazdığı **tek** şablon. Kalanlar kod
    /// varsayılanıyla (`isDefault: true`, `id: nil`) döner — sunucunun
    /// birleştirilmiş etkin görünümünün aynısı.
    static let templateReminderWhatsApp = "e4000000-0000-4000-8000-000000000001"

    // MARK: Şablonlar

    /// Kiracıya ait, kod varsayılanını ezen satırlar.
    static func tenantTemplates() -> [NotificationTemplate] {
        [
            NotificationTemplate(
                templateId: templateReminderWhatsApp,
                event: .appointmentReminder,
                channel: .whatsapp,
                locale: "tr",
                kind: .transactional,
                subject: nil,
                body: "Sayın {{customerName}}, {{appointmentAt}} tarihli {{serviceName}} randevunuzu hatırlatırız.",
                whatsappTemplateName: "randevu_hatirlatma",
                whatsappTemplateLanguage: "tr",
                whatsappVariables: ["customerName", "appointmentAt", "serviceName"],
                isActive: true,
                isDefault: false,
                variables: ["customerName", "appointmentAt", "serviceName"]
            )
        ]
    }

    /// Kod varsayılanları — sunucudaki `default-templates.ts`in istemci aynası.
    /// Yalnız mock için: canlıda bu satırlar sunucudan gelir.
    static func defaultTemplates() -> [NotificationTemplate] {
        NotificationEvent.selectable.flatMap { event -> [NotificationTemplate] in
            guard let definition = NotificationEventCatalog.definitions[event] else { return [] }
            return definition.channels.map { channel in
                NotificationTemplate(
                    templateId: nil,
                    event: event,
                    channel: channel,
                    locale: "tr",
                    kind: definition.kind,
                    subject: channel == .email ? "\(event.turkishName)" : nil,
                    // WhatsApp kanalının kod varsayılanı SMS gövdesine düşüyor
                    // (Ek M, düzeltilen hata #3): gerçek gönderimde zaten Meta'da
                    // onaylı template adı kullanılıyor.
                    body: defaultBody(for: event),
                    whatsappTemplateName: nil,
                    whatsappTemplateLanguage: nil,
                    whatsappVariables: [],
                    isActive: true,
                    isDefault: true,
                    variables: definition.variables
                )
            }
        }
    }

    private static func defaultBody(for event: NotificationEvent) -> String {
        switch event {
        case .appointmentConfirmation:
            return "Sayın {{customerName}}, {{branchName}} şubemizdeki {{appointmentAt}} tarihli {{serviceName}} randevunuz oluşturuldu."
        case .appointmentReminder:
            return "Sayın {{customerName}}, {{appointmentAt}} tarihli {{serviceName}} randevunuzu hatırlatırız."
        case .appointmentCancelled:
            return "Sayın {{customerName}}, {{appointmentAt}} tarihli randevunuz iptal edilmiştir."
        case .noShowFollowup:
            return "Sayın {{customerName}}, randevunuza katılamadınız. {{branchName}} şubemizden yeni bir randevu alabilirsiniz."
        case .packageBalance:
            return "Sayın {{customerName}}, {{packageName}} paketinizde {{remainingSessions}} seans hakkınız kaldı."
        case .packageExpiring:
            return "Sayın {{customerName}}, {{packageName}} paketiniz {{expiresAt}} tarihinde doluyor; {{remainingSessions}} seans hakkınız var."
        case .birthday:
            return "Sayın {{customerName}}, doğum gününüzü {{branchName}} olarak kutlarız."
        case .autoReply:
            return "{{message}}"
        case .staffInternal:
            return "{{subject}} — {{message}}"
        case .unknown:
            return ""
        }
    }

    // MARK: Tercihler

    /// Kiracı, doğum günü mesajını kapatmış (`channels: []`) ve randevu
    /// hatırlatmasında sessiz saati daraltmış. Kalan olaylar varsayılanda.
    static func tenantPreferences() -> [NotificationPreference] {
        [
            NotificationPreference(
                preferenceId: MockIDs.uuid(),
                branchId: nil,
                event: .appointmentReminder,
                kind: .transactional,
                channels: [.whatsapp, .email],
                quietHoursStart: "22:00",
                quietHoursEnd: "08:00",
                isDefault: false
            ),
            NotificationPreference(
                preferenceId: MockIDs.uuid(),
                branchId: nil,
                event: .birthday,
                kind: .marketing,
                channels: [],
                quietHoursStart: nil,
                quietHoursEnd: nil,
                isDefault: false
            ),
        ]
    }

    /// Kiracı satırı olmayan her olay için sentezlenen varsayılan.
    static func defaultPreference(for event: NotificationEvent) -> NotificationPreference {
        let definition = NotificationEventCatalog.definitions[event]
        return NotificationPreference(
            preferenceId: nil,
            branchId: nil,
            event: event,
            kind: definition?.kind ?? .transactional,
            channels: definition?.channels ?? [],
            quietHoursStart: "21:00",
            quietHoursEnd: "09:00",
            isDefault: true
        )
    }

    // MARK: Hatırlatma ayarları

    /// Nişantaşı şubesinin **kendi** override'ı var (24 + 4 saat); Bağdat
    /// caddesi kiracı varsayılanını kullanıyor. İki şubenin farklı olması,
    /// `isBranchOverride` ayrımının ekranda gerçekten sınanmasını sağlıyor.
    static func branchReminderOverrides() -> [String: [Int]] {
        [MockIDs.branchNisantasi: [24, 4]]
    }

    static let tenantReminderHours = [24, 2]

    // MARK: Mesaj günlüğü

    static func messages(at now: Date) -> [Message] {
        [
            message(
                id: messageReminderSent,
                customerId: MockCustomerSeed.ayse,
                channel: .whatsapp,
                event: .appointmentReminder,
                status: .read,
                to: "+90**********01",
                body: "Sayın Ayşe Yılmaz, yarın 10:00 randevunuzu hatırlatırız.",
                attempt: 1,
                scheduledFor: now.addingTimeInterval(-20 * 3_600),
                sentAt: now.addingTimeInterval(-20 * 3_600),
                deliveredAt: now.addingTimeInterval(-20 * 3_600 + 12),
                createdAt: now.addingTimeInterval(-26 * 3_600)
            ),
            message(
                id: messageConfirmationDelivered,
                customerId: MockCustomerSeed.zeynep,
                channel: .whatsapp,
                event: .appointmentConfirmation,
                status: .delivered,
                to: "+90**********03",
                body: "Sayın Zeynep Kaya, randevunuz oluşturuldu.",
                attempt: 1,
                scheduledFor: now.addingTimeInterval(-8 * 3_600),
                sentAt: now.addingTimeInterval(-8 * 3_600),
                deliveredAt: now.addingTimeInterval(-8 * 3_600 + 9),
                createdAt: now.addingTimeInterval(-8 * 3_600)
            ),
            // Ek M: engellenen mesaj ATILMIYOR, `skipped` yazılıyor. "Gitmedi mi,
            // hiç denendi mi?" sorusu cevaplanabilir kalmalı.
            message(
                id: messageBirthdaySkipped,
                customerId: MockCustomerSeed.mehmet,
                channel: .whatsapp,
                event: .birthday,
                status: .skipped,
                to: "+90**********02",
                body: "Sayın Mehmet Demir, doğum gününüzü kutlarız.",
                errorCode: APIErrorCode.optOut.rawValue,
                attempt: 0,
                scheduledFor: now.addingTimeInterval(-3 * 86_400),
                createdAt: now.addingTimeInterval(-3 * 86_400)
            ),
            message(
                id: messageReminderFailed,
                customerId: MockCustomerSeed.burak,
                channel: .whatsapp,
                event: .appointmentReminder,
                status: .failed,
                to: "+90**********04",
                body: "Sayın Burak Şahin, bugün 15:30 randevunuzu hatırlatırız.",
                errorCode: APIErrorCode.whatsappInvalidRecipient.rawValue,
                attempt: 1,
                scheduledFor: now.addingTimeInterval(-2 * 3_600),
                createdAt: now.addingTimeInterval(-5 * 3_600)
            ),
            // Alıcısı müşteri değil personel: `customerId` boş, `userId` dolu.
            message(
                id: messageStaffInternal,
                customerId: nil,
                userId: MockIDs.userOwner,
                channel: .email,
                event: .staffInternal,
                status: .sent,
                to: "o****@klinara.app",
                subject: "Gönderilemeyen hatırlatma",
                body: "Burak Şahin'in randevu hatırlatması gönderilemedi.",
                attempt: 1,
                scheduledFor: now.addingTimeInterval(-2 * 3_600 + 30),
                sentAt: now.addingTimeInterval(-2 * 3_600 + 32),
                createdAt: now.addingTimeInterval(-2 * 3_600 + 30)
            ),
        ]
    }

    private static func message(
        id: String,
        customerId: String?,
        userId: String? = nil,
        channel: NotificationChannel,
        event: NotificationEvent,
        status: MessageStatus,
        to: String,
        subject: String? = nil,
        body: String?,
        errorCode: String? = nil,
        attempt: Int,
        scheduledFor: Date,
        sentAt: Date? = nil,
        deliveredAt: Date? = nil,
        createdAt: Date
    ) -> Message {
        Message(
            id: id,
            customerId: customerId,
            userId: userId,
            channel: channel,
            event: event,
            status: status,
            to: to,
            subject: subject,
            body: body,
            errorCode: errorCode,
            attempt: attempt,
            scheduledFor: scheduledFor,
            sentAt: sentAt,
            deliveredAt: deliveredAt,
            createdAt: createdAt
        )
    }

    // MARK: Gelen kutusu

    static func inbox(at now: Date) -> [InboxItem] {
        [
            InboxItem(
                id: inboxAyse,
                customerId: MockCustomerSeed.ayse,
                from: "+90**********01",
                messageType: "text",
                body: "Merhaba, yarınki randevumu bir saat öne alabilir miyiz?",
                receivedAt: now.addingTimeInterval(-45 * 60),
                handledAt: nil
            ),
            // Tanınmayan numara: `customerId` yok. Sunucu bunu bilerek
            // eşleştirmiyor — yanlış müşteriye bağlamak yanlış kartı açardı.
            InboxItem(
                id: inboxUnknown,
                customerId: nil,
                from: "+90**********88",
                messageType: "text",
                body: "Fiyat listeniz var mı?",
                receivedAt: now.addingTimeInterval(-3 * 3_600),
                handledAt: nil
            ),
            InboxItem(
                id: inboxHandled,
                customerId: MockCustomerSeed.zeynep,
                from: "+90**********03",
                messageType: "image",
                body: nil,
                receivedAt: now.addingTimeInterval(-2 * 86_400),
                handledAt: now.addingTimeInterval(-2 * 86_400 + 1_800)
            ),
        ]
    }

    // MARK: İletişim izni

    static func optOuts(at now: Date) -> [OptOutRecord] {
        [
            // Mehmet tüm kanallarda ticari ileti almıyor — `messageBirthdaySkipped`
            // satırının sebebi bu. İki tohum birbirini AÇIKLAMALI, yoksa mock
            // veri gerçekte olamayacak bir durumu temsil eder.
            OptOutRecord(
                id: optOutMehmet,
                customerId: MockCustomerSeed.mehmet,
                channel: nil,
                kind: .marketing,
                source: .inboundStop,
                createdAt: now.addingTimeInterval(-10 * 86_400)
            )
        ]
    }

    // MARK: WhatsApp hesabı

    static func account(at now: Date) -> WhatsAppAccount {
        WhatsAppAccount(
            wabaId: "1029384756",
            phoneNumberId: "5647382910",
            businessPhone: "+902121234567",
            apiVersion: "v21.0",
            status: .active,
            accessTokenMasked: "••••••••aF3k",
            hasAppSecret: true,
            lastVerifiedAt: now.addingTimeInterval(-6 * 3_600),
            lastError: nil
        )
    }

    static func whatsAppTemplates(at now: Date) -> [WhatsAppTemplate] {
        [
            WhatsAppTemplate(
                name: "randevu_hatirlatma",
                language: "tr",
                category: "UTILITY",
                status: .approved,
                bodyVariableCount: 3,
                buttons: [
                    WhatsAppTemplateButton(type: "QUICK_REPLY", text: "Onayla"),
                    WhatsAppTemplateButton(type: "QUICK_REPLY", text: "İptal Et"),
                ],
                syncedAt: now.addingTimeInterval(-6 * 3_600)
            ),
            // Değişkensiz ve onaylı: test gönderiminin çalıştığı TEK şablon.
            WhatsAppTemplate(
                name: "baglanti_testi",
                language: "tr",
                category: "UTILITY",
                status: .approved,
                bodyVariableCount: 0,
                buttons: [],
                syncedAt: now.addingTimeInterval(-6 * 3_600)
            ),
            WhatsAppTemplate(
                name: "dogum_gunu",
                language: "tr",
                category: "MARKETING",
                status: .pending,
                bodyVariableCount: 2,
                buttons: [],
                syncedAt: now.addingTimeInterval(-6 * 3_600)
            ),
        ]
    }
}

File: klinara-android/app/src/main/java/com/klinara/android/services/notifications/MockNotificationsSeed.kt
package com.klinara.android.services.notifications

import com.klinara.android.services.contracts.ApiErrorCode
import com.klinara.android.services.integrations.InboxItem
import com.klinara.android.services.integrations.WhatsAppAccount
import com.klinara.android.services.integrations.WhatsAppAccountStatus
import com.klinara.android.services.integrations.WhatsAppTemplate
import com.klinara.android.services.integrations.WhatsAppTemplateButton
import com.klinara.android.services.integrations.WhatsAppTemplateStatus
import com.klinara.android.services.mock.MockCustomers
import com.klinara.android.services.mock.MockIds
import java.time.Duration
import java.time.Instant

/**
 * Mock bildirim verisinin başlangıç durumu — iOS `MockNotificationsSeed` paritesi.
 *
 * Tohum kasıtlı olarak **kurulumu yarım kalmış bir klinik**: bir hatırlatma okunmuş, biri
 * iletişim izni yüzünden `skipped` yazılmış, biri geçersiz numaraya takılmış, gelen kutusunda
 * işlenmemiş iki mesaj var (biri tanınmayan numaradan). Her şeyin yolunda olduğu bir tohum,
 * ekranların asıl zor durumlarını hiç göstermezdi.
 *
 * Müşteri kimlikleri [MockCustomers]'tan: gelen kutusundaki "Müşteri kartı" bağlantısı mock
 * grafiğinde gerçekten bir karta açılmalı.
 */
internal object MockNotificationsSeed {
    const val MESSAGE_REMINDER_READ = "e1000000-0000-4000-8000-000000000001"
    const val MESSAGE_CONFIRMATION_DELIVERED = "e1000000-0000-4000-8000-000000000002"
    const val MESSAGE_BIRTHDAY_SKIPPED = "e1000000-0000-4000-8000-000000000003"
    const val MESSAGE_REMINDER_FAILED = "e1000000-0000-4000-8000-000000000004"
    const val MESSAGE_STAFF_INTERNAL = "e1000000-0000-4000-8000-000000000005"

    const val INBOX_AYSE = "e2000000-0000-4000-8000-000000000001"
    const val INBOX_UNKNOWN = "e2000000-0000-4000-8000-000000000002"
    const val INBOX_HANDLED = "e2000000-0000-4000-8000-000000000003"

    const val OPT_OUT_MEHMET = "e3000000-0000-4000-8000-000000000001"

    val AYSE: String get() = MockCustomers.at(0).id
    val ZEYNEP: String get() = MockCustomers.at(1).id
    val FATMA: String get() = MockCustomers.at(FATMA_INDEX).id
    val MEHMET: String get() = MockCustomers.at(MEHMET_INDEX).id

    private const val FATMA_INDEX = 3
    private const val MEHMET_INDEX = 4

    /** Adlandırılmış argüman: tohum tablosu okunur kalsın, her sayı bir sabit olmasın. */
    private fun Instant.ago(
        days: Long = 0,
        hours: Long = 0,
        minutes: Long = 0,
        seconds: Long = 0,
    ): Instant = minus(Duration.ofDays(days).plusHours(hours).plusMinutes(minutes).plusSeconds(seconds))

    fun messages(now: Instant): List<Message> =
        listOf(
            Message(
                id = MESSAGE_REMINDER_READ,
                customerId = AYSE,
                channel = NotificationChannel.WhatsApp,
                event = NotificationEvent.AppointmentReminder,
                status = MessageStatus.Read,
                to = "+90**********33",
                body = "Sayın Ayşe Yılmaz, yarın 10:00 randevunuzu hatırlatırız.",
                attempt = 1,
                scheduledFor = now.ago(hours = 20),
                sentAt = now.ago(hours = 20),
                deliveredAt = now.ago(hours = 20, seconds = -12),
                createdAt = now.ago(hours = 26),
            ),
            Message(
                id = MESSAGE_CONFIRMATION_DELIVERED,
                customerId = ZEYNEP,
                channel = NotificationChannel.WhatsApp,
                event = NotificationEvent.AppointmentConfirmation,
                status = MessageStatus.Delivered,
                to = "+90**********34",
                body = "Sayın Zeynep Kaya, randevunuz oluşturuldu.",
                attempt = 1,
                scheduledFor = now.ago(hours = 8),
                sentAt = now.ago(hours = 8),
                deliveredAt = now.ago(hours = 8, seconds = -9),
                createdAt = now.ago(hours = 8),
            ),
            // Ek M: engellenen mesaj ATILMIYOR, `skipped` yazılıyor. Sebebi [optOuts]'taki kayıt —
            // iki tohum birbirini AÇIKLAMALI, yoksa mock gerçekte olamayacak bir durumu gösterir.
            Message(
                id = MESSAGE_BIRTHDAY_SKIPPED,
                customerId = MEHMET,
                channel = NotificationChannel.WhatsApp,
                event = NotificationEvent.Birthday,
                status = MessageStatus.Skipped,
                to = "+90**********37",
                body = "Sayın Mehmet Aslan, doğum gününüzü kutlarız.",
                errorCode = ApiErrorCode.OPT_OUT.wire,
                attempt = 0,
                scheduledFor = now.ago(days = 3),
                createdAt = now.ago(days = 3),
            ),
            Message(
                id = MESSAGE_REMINDER_FAILED,
                customerId = FATMA,
                channel = NotificationChannel.WhatsApp,
                event = NotificationEvent.AppointmentReminder,
                status = MessageStatus.Failed,
                to = "+90**********36",
                body = "Sayın Fatma Şahin, bugün 15:30 randevunuzu hatırlatırız.",
                errorCode = ApiErrorCode.WHATSAPP_INVALID_RECIPIENT.wire,
                attempt = 1,
                scheduledFor = now.ago(hours = 2),
                createdAt = now.ago(hours = 5),
            ),
            // Alıcısı müşteri değil personel: `customerId` boş, `userId` dolu.
            Message(
                id = MESSAGE_STAFF_INTERNAL,
                userId = MockIds.USER_MANAGER,
                channel = NotificationChannel.Email,
                event = NotificationEvent.StaffInternal,
                status = MessageStatus.Sent,
                to = "a****@klinara.app",
                subject = "Gönderilemeyen hatırlatma",
                body = "Fatma Şahin'in randevu hatırlatması gönderilemedi.",
                attempt = 1,
                scheduledFor = now.ago(hours = 2, seconds = -30),
                sentAt = now.ago(hours = 2, seconds = -32),
                createdAt = now.ago(hours = 2, seconds = -30),
            ),
        )

    fun inbox(now: Instant): List<InboxItem> =
        listOf(
            InboxItem(
                id = INBOX_AYSE,
                customerId = AYSE,
                from = "+90**********33",
                messageType = "text",
                body = "Merhaba, yarınki randevumu bir saat öne alabilir miyiz?",
                receivedAt = now.ago(minutes = 45),
            ),
            // Tanınmayan numara: sunucu bilerek eşleştirmiyor — yanlış müşteriye bağlamak
            // yanlış kartı açardı.
            InboxItem(
                id = INBOX_UNKNOWN,
                from = "+90**********88",
                messageType = "text",
                body = "Fiyat listeniz var mı?",
                receivedAt = now.ago(hours = 3),
            ),
            InboxItem(
                id = INBOX_HANDLED,
                customerId = ZEYNEP,
                from = "+90**********34",
                messageType = "image",
                receivedAt = now.ago(days = 2),
                handledAt = now.ago(days = 2, minutes = -30),
            ),
        )

    const val TEMPLATE_REMINDER_WHATSAPP = "e4000000-0000-4000-8000-000000000001"

    /**
     * Kiracının kendi metnini yazdığı TEK şablon — ve kod varsayılanı OLMAYAN bir kanalda
     * (WhatsApp): sunucu onu listenin sonuna ekliyor, mock da öyle.
     */
    fun tenantTemplates(): List<NotificationTemplate> =
        listOf(
            NotificationTemplate(
                templateId = TEMPLATE_REMINDER_WHATSAPP,
                event = NotificationEvent.AppointmentReminder,
                channel = NotificationChannel.WhatsApp,
                kind = NotificationKind.Transactional,
                body = "Sayın {{customerName}}, {{appointmentAt}} tarihli {{serviceName}} randevunuzu hatırlatırız.",
                whatsappTemplateName = "randevu_hatirlatma",
                whatsappTemplateLanguage = "tr",
                whatsappVariables = listOf("customerName", "appointmentAt", "serviceName"),
                variables = listOf("customerName", "appointmentAt", "serviceName"),
            ),
        )

    /**
     * Kod varsayılanları — sunucudaki `default-templates.ts`'in `templates` anahtarlarıyla
     * **birebir aynı kanallar** (çoğu olayda SMS + e-posta; WhatsApp yalnız otomatik yanıtta).
     *
     * iOS mock'u varsayılanı olayın TÜM kanallarından üretiyordu ve canlıda hiç görünmeyen
     * WhatsApp satırlarını gösteriyordu; sunucunun gerçek davranışı saklanıyordu.
     */
    fun defaultTemplates(): List<NotificationTemplate> =
        DEFAULT_BODIES.flatMap { (event, byChannel) ->
            byChannel.map { (channel, body) ->
                NotificationTemplate(
                    event = event,
                    channel = channel,
                    kind = NotificationEventCatalog.kind(event),
                    subject = event.turkishName.takeIf { channel == NotificationChannel.Email },
                    body = body,
                    isDefault = true,
                    variables = NotificationEventCatalog.placeholders(body),
                )
            }
        }

    private val DEFAULT_BODIES: List<Pair<NotificationEvent, List<Pair<NotificationChannel, String>>>> =
        listOf(
            NotificationEvent.AppointmentConfirmation to
                smsOnly("Sayın {{customerName}}, {{appointmentAt}} randevunuz oluşturuldu. {{branchName}}"),
            NotificationEvent.AppointmentReminder to
                smsOnly("Sayın {{customerName}}, {{appointmentAt}} randevunuzu hatırlatırız. {{branchName}}"),
            NotificationEvent.AppointmentCancelled to
                smsOnly("Sayın {{customerName}}, {{appointmentAt}} randevunuz iptal edilmiştir. {{branchName}}"),
            NotificationEvent.NoShowFollowup to
                listOf(
                    NotificationChannel.Sms to
                        "Sayın {{customerName}}, randevunuza katılamadınız. " +
                            "{{branchName}} olarak yeni randevu için bekleriz.",
                ),
            NotificationEvent.PackageBalance to
                smsOnly("Sayın {{customerName}}, {{packageName}} paketinizde {{remainingSessions}} seans kaldı."),
            NotificationEvent.PackageExpiring to
                smsOnly("Sayın {{customerName}}, {{packageName}} paketiniz {{expiresAt}} tarihinde doluyor."),
            NotificationEvent.Birthday to
                listOf(NotificationChannel.Sms to "Sayın {{customerName}}, doğum gününüzü kutlarız! {{branchName}}"),
            NotificationEvent.AutoReply to
                listOf(NotificationChannel.WhatsApp to "{{message}}", NotificationChannel.Sms to "{{message}}"),
            NotificationEvent.StaffInternal to listOf(NotificationChannel.Email to "{{message}}"),
        )

    /** Müşteri olaylarının varsayılan gövdesi yalnız SMS'te durur; WhatsApp metni Meta'dan gelir. */
    private fun smsOnly(body: String) = listOf(NotificationChannel.Sms to body)

    /**
     * Kiracı doğum günü mesajını kapatmış (`channels: []`) ve randevu hatırlatmasında sessiz
     * saati daraltmış. Kalan olaylar sunucunun sentezlediği varsayılanla gelir.
     */
    fun tenantPreferences(): List<NotificationPreference> =
        listOf(
            NotificationPreference(
                preferenceId = "e5000000-0000-4000-8000-000000000001",
                event = NotificationEvent.AppointmentReminder,
                kind = NotificationKind.Transactional,
                channels = listOf(NotificationChannel.WhatsApp, NotificationChannel.Sms),
                quietHoursStart = "22:00",
                quietHoursEnd = "08:00",
            ),
            NotificationPreference(
                preferenceId = "e5000000-0000-4000-8000-000000000002",
                event = NotificationEvent.Birthday,
                kind = NotificationKind.Marketing,
                channels = emptyList(),
            ),
        )

    fun account(now: Instant): WhatsAppAccount =
        WhatsAppAccount(
            wabaId = "1029384756",
            phoneNumberId = "5647382910",
            businessPhone = "+902121234567",
            apiVersion = "v21.0",
            status = WhatsAppAccountStatus.Active,
            accessTokenMasked = "••••••••aF3k",
            hasAppSecret = true,
            lastVerifiedAt = now.ago(hours = 6),
        )

    /** Biri değişkenli, biri test edilebilir (değişkensiz + onaylı), biri onay bekliyor. */
    fun whatsAppTemplates(now: Instant): List<WhatsAppTemplate> =
        listOf(
            WhatsAppTemplate(
                name = "randevu_hatirlatma",
                language = "tr",
                category = "UTILITY",
                status = WhatsAppTemplateStatus.Approved,
                bodyVariableCount = 3,
                buttons =
                    listOf(
                        WhatsAppTemplateButton("QUICK_REPLY", "Onayla"),
                        WhatsAppTemplateButton("QUICK_REPLY", "İptal Et"),
                    ),
                syncedAt = now.ago(hours = 6),
            ),
            WhatsAppTemplate(
                name = "baglanti_testi",
                language = "tr",
                category = "UTILITY",
                status = WhatsAppTemplateStatus.Approved,
                syncedAt = now.ago(hours = 6),
            ),
            // Aynı ad başka dilde: test ekranı seçimi ad + dil ile yapmalı (iOS yalnız ada bakıyordu).
            WhatsAppTemplate(
                name = "baglanti_testi",
                language = "en",
                category = "UTILITY",
                status = WhatsAppTemplateStatus.Approved,
                syncedAt = now.ago(hours = 6),
            ),
            WhatsAppTemplate(
                name = "dogum_gunu",
                language = "tr",
                category = "MARKETING",
                status = WhatsAppTemplateStatus.Pending,
                bodyVariableCount = 2,
                syncedAt = now.ago(hours = 6),
            ),
        )

    /**
     * Mehmet tüm kanallarda ticari ileti almıyor — [MESSAGE_BIRTHDAY_SKIPPED] satırının sebebi.
     *
     * A4.2'deki mock tohumu "bilerek boş değil" diyordu ama boştu; kapalı hâl ancak burada
     * sürülebilir hâle geldi.
     */
    fun optOuts(now: Instant): List<OptOutRecord> =
        listOf(
            OptOutRecord(
                id = OPT_OUT_MEHMET,
                customerId = MEHMET,
                channel = null,
                kind = "marketing",
                source = OptOutSource.InboundStop,
                createdAt = now.ago(days = 10),
            ),
        )
}
